packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+        if (
+            changes.has('open') &&
+            (typeof changes.get('open') !== 'undefined' || this.open)
+        ) {
+            this.manageOpen(changes.get('open'));
+        }
+        if (changes.has('trigger')) {
+            const [id, interaction] = this.trigger?.split('@') || [];
+            this.elementResolver.selector = id ? `#${id}` : '';
+            this.triggerInteraction = interaction as
+                | 'click'
+                | 'longpress'
+                | 'hover'
+                | undefined;
+        }
+        const oldTrigger = this.triggerElement as HTMLElement;
+        if (changes.has(elementResolverUpdatedSymbol)) {
+            this.triggerElement = this.elementResolver.element;
+            this.manageTriggerElement(oldTrigger);
+        }
+        if (changes.has('triggerElement')) {
+            this.manageTriggerElement(changes.get('triggerElement'));
+        }
+    }
+
+    protected override updated(changes: PropertyValues): void {
+        super.updated(changes);
+        if (changes.has('placement')) {
+            if (this.placement) {
+                this.dialogEl.setAttribute('actual-placement', this.placement);
+            } else {
+                this.dialogEl.removeAttribute('actual-placement');
+            }
+            if (this.open && typeof changes.get('placement') !== 'undefined') {
+                this.placementController.resetOverlayPosition();
+            }
+        }
+    }
+
+    protected renderContent(): TemplateResult {
+        return html`
+            <div part="content">
+                <slot @slotchange=${this.handleSlotchange}></slot>
+            </div>
+        `;
+    }
+
+    protected renderDialog(): TemplateResult {
+        return html`
+            <dialog
+                class="dialog"
+                part="dialog"
+                @close=${this.handleBrowserClose}
+                @cancel=${this.handleBrowserClose}
+                @beforetoggle=${this.handleBeforetoggle}
+                style=${styleMap({
+                    '--swc-overlay-open-count':
+                        OverlayBase.openCount.toString(),
+                    translate: this.hasUpdated ? null : '-999em -999em',
+                })}
+            >
+                ${this.renderContent()}
+            </dialog>
+        `;
+    }
+
+    protected renderPopover(): TemplateResult {
+        const hasPopoverAttribute = 'popover' in this;
+        const popoverValue = hasPopoverAttribute
+            ? this.popoverValue
+            : undefined;
+        return html`
+            <div
+                class="dialog"
+                part="dialog"
+                popover=${ifDefined(popoverValue)}
+                @beforetoggle=${this.handleBeforetoggle}
+                @close=${this.handleBrowserClose}
+                style=${styleMap({
+                    '--swc-overlay-z-index': (
+                        1000 + OverlayBase.openCount
+                    ).toString(),
```

Comment:
FMI, could you share why  Dialog & Popover use similar but slightly-different mechanisms for setting z-index?

Reply:
Error!